src/Microsoft.Azure.EventHubs/EventHubsDiagnosticSource.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.EventHubs
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Diagnostic Source util for "Microsoft.Azure.EventHubs" source.
+    /// </summary>
+    internal static class EventHubsDiagnosticSource
+    {
+        public const string DiagnosticSourceName = "Microsoft.Azure.EventHubs";
+
+        public const string SendActivityName = DiagnosticSourceName + ".Send";
+        public const string SendActivityStartName = SendActivityName + ".Start";
+        public const string SendActivityExceptionName = SendActivityName + ".Exception";
+
+        public const string ReceiveActivityName = DiagnosticSourceName + ".Receive";
+        public const string ReceiveActivityStartName = ReceiveActivityName + ".Start";
+        public const string ReceiveActivityExceptionName = ReceiveActivityName + ".Exception";
+
+        internal static readonly DiagnosticListener DiagnosticListener = new DiagnosticListener(DiagnosticSourceName);
+
+        internal static bool IsEnabled => DiagnosticListener.IsEnabled();
```

Comment:
it means that isEnabled decision is made when EH first called and never changes. What if listener subscribes after that?
IsEnabled should not be cached.

Reply:
good point, changed